src/qibolab/platforms/platform.py

```diff
@@ -150,3 +154,21 @@ class ExecutionParameters:
     fast_reset: bool = False
     acquisition_type: AcquisitionType = AcquisitionType.DISCRIMINATION
     averaging_mode: AveragingMode = AveragingMode.SINGLESHOT
+
+    def get_results_type(self):
+        """Returns corresponding results class"""
```

Comment:
Another way to do this that could improve readability would be to have a global mapping like
```py
RESULTS_TYPE = {
    AveragingMode.CYCLIC: {
        AcquisitionType.INTEGRATION: AveragedIntegratedResults,
        AcquisitionType.RAW: AveragedRawWaveformResults,
        AcquisitionType.DISCRIMINATION: AveragedStateResults,
    },
    AveragingMode.SINGLESHOT: {
        AcquisitionType.INTEGRATION: IntegratedResults,
        AcquisitionType.RAW: RawWaveformResults,
        AcquisitionType.DISCRIMINATION: StateResults,
    }
```
and then this function would look like
```py
def get_results_type(self):
    return RESULTS_TYPE[self.averaging_mode][self.acquisition_type]
```
Also, this is a getter that takes no argument, so consider making it a property:
```py
@property
def results_type(self):
    return RESULTS_TYPE[self.averaging_mode][self.acquisition_type]
```

Reply:
That's definitely better, thank you